Added blackbox features and made changes to the design

alexandru-grigoras opened the pull request:
The logs are now only opened/closed once per run.
Added timestamps to the run names.
Added the option to delete runs.
Better handling of graphs and logs.
Made some UI changes to differentiate between the various types of data displayed.

eoswald commented:
```
Traceback (most recent call last):
  File "server.py", line 64, in <module>
    f = open('static/logs/' + str(s) + ".csv",'w')
IOError: [Errno 2] No such file or directory: 'static/logs/velocity.csv'
```
This should be able to handle creating new files, right?

eoswald commented on server.py:
```diff
@@ -79,35 +84,69 @@ def sensor_log(cmd, data):
+        for s in logged_sensors:
+            f = open('static/logs/' + str(s) + ".csv",'r')
+            lines = f.readlines()
+            f.close()
```
```
with open('static/logs/' + str(s) + ".csv",'r') as f:
    lines = f.readline()
```

alexandru-grigoras replied:
1. The creation of the 'logs' directory will be fixed in the next commit.

2. I don't think that could work, actually, because I want to read from the file, close it, then write back to it. If I use the "with" statement, Python handles the closing for me, so I either need to change the scope of 'lines' or I will be reading and writing from the same file at the same time, which I believe work but probably shouldn't do.

eoswald replied:
I'm fairly certain Python variables are function scoped. So that should work.

eoswald commented:
So how does this work? Runs logs are only updated for the duration of a run but sensor logs are always being updated?

eoswald commented on server.py:
```diff
@@ -79,35 +84,69 @@ def sensor_log(cmd, data):
-            f = open('static/data/' + str(s) + ".csv",'a')
-            f.write('run_' + str(pod_runs) + ',' + str(time_passed) + ',' + str(data[s]) + '\n')
-            f.close()
+            output_files[s].close()
```
You can use seek() instead of just closing and reopening the files

alexandru-grigoras replied:
I generally need to close them in order to write the buffer to file, so that the charts can be displayed.

eoswald commented:
Seems like a lot of work to avoid having all the sensor values comma separated in one file.

alexandru-grigoras commented:
Fixed sensor log folder creation problem
Removed manual controls
Fixed formatting issues for various screen sizes
Modified elements to better fit to screen
Changed the thermometers' style
Replaced the globe with a diamond shape that represents the pod

eoswald commented:
Looks really good. Merging
